cookbooks/aws-parallelcluster-slurm/recipes/update_head_node.rb

```diff
@@ -15,7 +15,7 @@
 # OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
 # limitations under the License.
 
-unless ::FileUtils.identical?(node['cluster']['previous_cluster_config_path'], node['cluster']['cluster_config_path'])
+unless ::File.exist?(node['cluster']['previous_cluster_config_path']) && ::FileUtils.identical?(node['cluster']['previous_cluster_config_path'], node['cluster']['cluster_config_path'])
```

Comment:
[The config is fetched](https://github.com/aws/aws-parallelcluster-cookbook/blob/develop/cookbooks/aws-parallelcluster-config/recipes/update.rb#L19) before the execution of this recipe.
If the conditional statement here is done at compile time, it will always evaluate to `false`

Reply:
i was thinking that it would work when run as part of the first cluster update because fetch_config is run in a few recipes that would be executed during cluster creation. But the update flag isn't specified as part of those usages, so you're right.

Reply:
fixed by adding `not_if` guards to all the resources contained in the recipe. it's an admittedly ugly solution so let me know if you know a less repetitive way to do it.